Symfony/CS/Fixer/Contrib/Php4ConstructorFixer.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+/*
+ * This file is part of the Symfony CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Matteo Beccati <matteo@beccati.com>
+ */
+class Php4ConstructorFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        foreach ($tokens->findGivenKind(T_CLASS) as $index => $token) {
+            // Is it inside a namespace?
+            $nspIndex = $tokens->getPrevTokenOfKind($index, array(array(T_NAMESPACE, 'namespace')));
+            if ($nspIndex) {
+                $nspName = $tokens->getNextMeaningfulToken($nspIndex); // Name
+                $nspBrace = $tokens->getNextMeaningfulToken($nspName); // { or ;
+
+                if ($tokens[$nspBrace]->equals(';')) {
+                    // Class is inside a (non-block) namespace
+                    continue;
+                } else {
+                    if ($index < $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $nspBrace)) {
+                        // Class is inside a block namespace
+                        continue;
+                    }
+                }
+            }
+
+            $classNameIndex = $tokens->getNextMeaningfulToken($index);
+            $className = $tokens[$classNameIndex]->getContent();
+            $classStart = $tokens->getNextTokenOfKind($classNameIndex, array('{'));
+            $classEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $classStart);
+
+            $p4 = $this->findFunction($tokens, $className, $classStart, $classEnd);
+
+            if (!$p4) {
+                // No PHP4-constructor!
+                continue;
+            }
+
+            $p5 = $this->findFunction($tokens, '__construct', $classStart, $classEnd);
+
+            if (!$p5) {
+                // No PHP5-constructor, we can rename the old one
+                $tokens[$p4['nameIdx']]->setContent('__construct');
+                continue;
+            }
+
+            // Does the PHP4-constructor call __construct?
+            if (false !== stripos($p4['body'], '$this->__construct')) {
```

Comment:
It's just an extra short proof of bug. When `__construct` is an constructor, one may have in `Foo`  anything he wants. Not just an error like here - it just to let you see the error. One may contain here anything - eg whole class description, JS code, etc.
And you must not remove it.

One day we were doing everything on text, which lead us to "fix" things like
`if (1) echo 1; else if (2) echo 2;`
into 
`if (1) echo 1; elseif (2) echo 2;`
which is great, but we did it even inside string, like:
`$jsCode = '<script>if (1) echo 1; else if (2) echo 2;`</script>';`

which was horrible error (not only changed content of string, but also injected a bug (there is no `elseif` in JS)).


Now, you even remove whole method :(



This tool is supposed to run on code to fix it's CS. So, the code may be extra ugly and meaningless. And the tool must not break it.